test/e2e/self_hosted.go

```diff
@@ -389,13 +389,15 @@ func SelfHostedSpec(ctx context.Context, inputGetter func() SelfHostedSpecInput)
 	})
 
 	AfterEach(func() {
+		//dumpNodesAndPods(ctx, specName, input.BootstrapClusterProxy, input.ArtifactFolder, namespace, cancelWatches, clusterResources.Cluster, input.E2EConfig.GetIntervals, input.SkipCleanup)
 		if selfHostedNamespace != nil {
 			// Dump all Cluster API related resources to artifacts before pivoting back.
 			framework.DumpAllResources(ctx, framework.DumpAllResourcesInput{
 				Lister:    selfHostedClusterProxy.GetClient(),
 				Namespace: namespace.Name,
 				LogPath:   filepath.Join(input.ArtifactFolder, "clusters", clusterResources.Cluster.Name, "resources"),
 			})
+			dumpNodesAndPods(ctx, specName, input.BootstrapClusterProxy, input.ArtifactFolder, namespace, cancelWatches, clusterResources.Cluster, input.E2EConfig.GetIntervals, input.SkipCleanup)
```

Comment:
```suggestion
			dumpNodesAndPods(ctx, specName, selfHostedClusterProxy, input.ArtifactFolder, namespace, cancelWatches, clusterResources.Cluster, input.E2EConfig.GetIntervals, input.SkipCleanup)
```

We're not getting the correct resources dumped (though we are getting the failure :slightly_smiling_face:). This should be the proxy for the selfHostedCluster which is where the resources should be found at this point in time.

Reply:
Added via 3013d60e1a855f7672d9ac9eeb4aaf6b411fbec2